src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java

```diff
@@ -86,37 +117,54 @@ public ClusterTestHarness() {
   }
 
   public ClusterTestHarness(int numBrokers, boolean withSchemaRegistry) {
-    // 1 port per broker + ZK + possibly SchemaReg + REST server
-    int numPorts = numBrokers + 3;
-    ports = new ArrayDeque<Integer>();
-    for (Object portObj : JavaConversions.asJavaList(TestUtils.choosePorts(numPorts))) {
-      ports.add((Integer) portObj);
-    }
-    zkPort = ports.remove();
-    zkConnect = String.format("localhost:%d", zkPort);
+    this.numBrokers = numBrokers;
+    this.withSchemaRegistry = withSchemaRegistry;
+
+    schemaRegProperties = new Properties();
+    restProperties = new Properties();
+  }
 
-    configs = new Vector<Properties>();
-    bootstrapServers = "";
+  public Properties overrideBrokerProperties(int i, Properties props) {
```

Comment:
It's a hook for subclasses to allow them to override broker properties only for their tests. See, e.g. `ProducerTopicAutoCreationTest`.